src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3949,19 +3950,54 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
 
                 MemberResolutionResult<MethodSymbol> memberResolutionResult;
                 ImmutableArray<MethodSymbol> candidateConstructors;
-                if (TryPerformConstructorOverloadResolution(
-                    initializerType,
-                    analyzedArguments,
-                    WellKnownMemberNames.InstanceConstructorName,
-                    errorLocation,
-                    false, // Don't suppress result diagnostics
-                    diagnostics,
-                    out memberResolutionResult,
-                    out candidateConstructors,
-                    allowProtectedConstructorsOfBaseType: true))
+                bool found;
+                MethodSymbol resultMember;
+                ImmutableArray<int> argsToParamsOpt;
+                bool isExpanded;
+                if (constructor is SynthesizedRecordCopyCtor)
```

Comment:
>if (constructor is SynthesizedRecordCopyCtor) [](start = 16, length = 45)

I think this code belongs in SynthesizedRecordCopyCtor.GenerateMethodBodyStatements. There is nothing to bind for that constructor, everything is synthesized, including the initializer. We also shouldn't call this function for SynthesizedRecordCopyCtor, which I think can easily be achieved by detecting it in MethodCompiler.BindImplicitConstructorInitializerIfAny.

Reply:
Using MethodCompiler.GenerateBaseParameterlessConstructorInitializer as an example might help

---
In reply to: [438502981](https://github.com/dotnet/roslyn/pull/45022#discussion_r438502981) [](ancestors = 438502981)

Reply:
I refactored the code out to a method similar to `GenerateBaseParameterlessConstructorInitializer`, but not into `SynthesizedRecordCopyCtor.GenerateMethodBodyStatements`.
I think that is more consistent with existing flow and overall was simpler. Please file an issue if you strongly think it should be refactored further.

---
In reply to: [438516676](https://github.com/dotnet/roslyn/pull/45022#discussion_r438516676) [](ancestors = 438516676,438502981)